finagle-http/src/test/scala/com/twitter/finagle/http/codec/ResponseConformanceFilterTest.scala

```diff
@@ -78,6 +79,69 @@ class ResponseConformanceFilterTest extends FunSuite {
     intercept[ReaderDiscarded] { Await.result(res.writer.write(Buf.Empty), 5.seconds) }
   }
 
+  test("response with status code {1xx, 204 and 304} must not have a message body nor Content-Length header field") {
+    def validate(status: Status) = {
+      val res = Response(Version.Http11, status)
+      val response = fetchResponse(res)
+
+      assert(response.status == status)
+      assert(response.getContent.readableBytes() == 0)
+      assert(!response.isChunked)
+      assert(response.headers().get(Fields.ContentLength) == null)
```

Comment:
Try to use `.headerMap` to remove netty types from the testing. It's also nice not to look for `null` in scala code. There are a few more instances of this below as well.